sycl/doc/EnvironmentVariables.md

```diff
@@ -65,7 +65,7 @@ Possible values of "device_type" are:
 - acc
 - \*
 
-Device_num is an integer that indexes the enumeration of devices from the sycl-ls utility tool, where the first device in that enumeration has index zero in each backend. For example, SYCL_DEVICE_FILTER=2 will return all devices with index '2' from all different backends. If multiple devices satisfy this device number (e.g., GPU and CPU devices can be assigned device number '2'), then default_selector will choose the device with the highest heuristic point. 
+Device_num is an integer that indexes the enumeration of devices from the sycl-ls utility tool, where the first device in that enumeration has index zero in each platform. For example, SYCL_DEVICE_FILTER=2 will return all devices with index '2' from all different platforms. If multiple devices satisfy this device number (e.g., GPU and CPU devices can be assigned device number '2'), then default_selector will choose the device with the highest heuristic point. 
```

Comment:
If devices reported in multiple platforms (like Intel GPUs on Windows currently) then restarting numbering by platforms will yield indistinguishable IDs for those device. i.e. `level_zero:gpu:0` 

I think we need to continue numbering by backend and fix the `sycl-ls` instead.

Reply:
Are you talking about the case where users install different device drivers?
In that case, the user should use SYCL_DEVICE_ALLOWLIST.
SYCL_DEVICE_FILTER is not designed to cover that usage case.
If we use unique device-id per backend, the device numbering becomes a little awkward and error-prone for vastly common use cases.
     
 backend-unique                         platform-unique (current)
---------------------------------------------------------
OpenCL:CPU:0                              OpenCL:CPU:0
OpenCL:CPU:1                              OpenCL:CPU:1
OpenCL:ACC:2                              OpenCL:ACC:0
OpenCL:ACC:3                              OpenCL:ACC:1
OpenCL:GPU:4                              OpenCL:GPU:0
OpenCL:GPU:5                              OpenCL:GPU:1

Currently, we print the driver version when listing devices from 'sycl-ls'.
So, users can easily use SYCL_DEVICE_ALLOWLIST using that driver information.

@romanovvlad @vladimirlaz 
What do you think about this?

Reply:
> Are you talking about the case where users install different device drivers?

I am talking about the case when https://spec.oneapi.com/level-zero/latest/core/api.html#zedriverget returns multiple drivers (platforms in SYCL notion), each containing a GPU. We need to make it possible to select specific device with just the SYCL_DEVICE_FILTER.

Reply:
@smaslov-intel ,
Thanks for the link. I think I got it now.
I understand that we treat each instance of the driver as separate platforms.
So, zeDriverGet() will return two different platforms if we have multiple GPU devices.
I will make the device-id unique per backend as we originally planned.